modules/downloadmode.py

```diff
@@ -0,0 +1,32 @@
+from tidal import *
+import textwindow
+from tidal_helpers import reboot_bootloader
+
+# Note, this intentionally doesn't inherit App or TextApp to limit dependencies
+# because it is on the critical path from the Recovery Menu.
+# But it acts sufficiently like it does, to satisfy the app launcher
+class DownloadMode:
```

Comment:
I probably wouldn't have bothered making this an object, since it isn't also shown in the app_launcher the act-like-an-App stuff isn't really necessary. It could be a single static function? Given it seems to only be 5 non-boilerplate lines of code I'd be inclined to just roll it in to bootmenu.py like erase_storage is (although that fn is a little long).

Otherwise though, logic looks fine (assuming the usb code does the trick) and no objections to having this in the recovery menu.